### PR TITLE
feat(generator): boosties v2

### PR DESCRIPTION
**Pull request by P-Yevhenii**

### CODE REVIEW

File: contracts/tokenomics/generator/src/migration.rs [hunk 1]
```diff
@@ -119,6 +155,7 @@ pub fn migrate_configs_from_v100(
         factory: addr_validate_to_lower(deps.api, &msg.factory.clone().unwrap())?,
         generator_controller: None,
         voting_escrow: None,
+        voting_escrow_delegation: None,
```
Comment (ShadoySV) [inline contracts/tokenomics/generator/src/migration.rs:158]: Set it from ```MigrateMsg```. The same below

File: contracts/tokenomics/generator/src/state.rs [hunk 2]
```diff
@@ -185,9 +185,17 @@ pub(crate) fn update_virtual_amount(
     let mut user_vp = Uint128::zero();
     let mut total_vp = Uint128::zero();
 
-    if let Some(voting_escrow) = &voting_escrow {
-        user_vp = get_voting_power(&querier, voting_escrow, account)?;
-        total_vp = get_total_voting_power(&querier, voting_escrow)?;
+    if let Some(voting_escrow) = &cfg.voting_escrow {
+        if let Some(voting_delegation) = &cfg.voting_escrow_delegation {
```
Comment (ShadoySV) [inline contracts/tokenomics/generator/src/state.rs:189]: Use original voting power if the voting escrow delegation contract isn't set

**Review by ShadoySV — CHANGES_REQUESTED** [review]

**Review by ShadoySV — APPROVED** [review]

File: contracts/tokenomics/generator/src/contract.rs [hunk 3]
```diff
@@ -563,14 +569,19 @@ pub fn execute_update_config(
...
-        config.voting_escrow = Some(addr_validate_to_lower(deps.api, voting_escrow.as_str())?);
-    }
-
     if let Some(generator_limit) = checkpoint_generator_limit {
         config.checkpoint_generator_limit = Some(generator_limit);
     }
 
+    if let Some(voting_escrow_delegation) = voting_escrow_delegation {
+        config.voting_escrow_delegation =
+            Option::from(addr_validate_to_lower(deps.api, &voting_escrow_delegation)?);
```
Comment (epanchee) [inline contracts/tokenomics/generator/src/contract.rs:578]:
```suggestion
            Some(addr_validate_to_lower(deps.api, &voting_escrow_delegation)?);
```

File: contracts/tokenomics/generator/src/contract.rs [hunk 4]
```diff
@@ -563,14 +569,19 @@ pub fn execute_update_config(
...
+    }
+
+    if let Some(voting_escrow) = voting_escrow {
+        config.voting_escrow = Option::from(addr_validate_to_lower(deps.api, &voting_escrow)?);
```
Comment (epanchee) [inline contracts/tokenomics/generator/src/contract.rs:582]:
```suggestion
        config.voting_escrow = Some(addr_validate_to_lower(deps.api, &voting_escrow)?);
```

**Review by epanchee — APPROVED** [review]

**Review by ShadoySV — CHANGES_REQUESTED** [review]
Dependencies with ```feat/escrow_delegation``` should be replaced with ```main``` when them is merged

**Review by ShadoySV — APPROVED** [review]

**Pull request merged**